src/db/db.h

```diff
@@ -4,6 +4,12 @@
 #include <string.h>
 #include <strings.h>
 
+#ifndef __APPLE__
```

Comment:
this argument swap issue is terrifying. there has got to be a better way (not involving the ifdef around *all* compare functions above).

Reply:
Yes I know, I want just to reach a working/compiling solution and then beutify. 😄